SCClassLibrary/scide_scqt/ScIDE.sc

```diff
@@ -637,6 +637,7 @@ Document {
 	save { |docPath|
 		docPath = docPath ? path;
 		if(docPath.isNil) { MethodError("Document saved requires specified path", this).throw; };
+		if(docPath.dirname.pathMatch.size ==  0) { error("Document save failed as directory does not exist.") };
```

Comment:
would we want to throw that error? Not sure myself, just wondering.

Reply:
As I wrote in the comment, I think the errors should actually be thrown on the c++ side, as opposed to trying to manually check for a limited set of conditions.

Reply:
As to whether it should be thrown or not I can imagine people may want to catch it.

Reply:
I feel this should throw an error, more below.